mbed/mbed.py

```diff
@@ -774,17 +791,41 @@ def getbranch(rev='HEAD'):
             branch = "master"
         return branch if branch != "HEAD" else ""
 
-    # Finds refs (local or remote branches). Will match rev if specified
-    def getrefs(rev=None, ret_rev=False):
+    # Get all refs
+    def getrefs():
+        try:
+            return pquery([git_cmd, 'show-ref', '--dereference']).strip().splitlines()
+        except ProcessException:
+            return []
+
+    # Finds branches (local or remote). Will match rev if specified
+    def getbranches(rev=None, ret_rev=False):
         result = []
-        lines = pquery([git_cmd, 'show-ref']).strip().splitlines()
-        for line in lines:
-            m = re.match(r'^(.+)\s+(.+)$', line)
+        refs = Git.getrefs()
+        for ref in refs:
+            m = re.match(r'^(.+)\s+(.+)$', ref)
             if m and (not rev or m.group(1).startswith(rev)):
                 if re.match(r'refs\/(heads|remotes)\/', m.group(2)): # exclude tags
                     result.append(m.group(1) if ret_rev else re.sub(r'refs\/(heads|remotes)\/', '', m.group(2)))
         return result
 
+    # Finds tags. Will match rev if specified
+    def gettags(rev=None):
+        tags = []
+        refs = Git.getrefs()
+        for ref in refs:
+            m = re.match(r'^(.+)\s+(.+)$', ref)
+            if m and (not rev or m.group(1).startswith(rev)):
+                if re.match(r'refs\/tags\/', m.group(2)): # only tags
+                    t = re.sub(r'refs\/tags\/', '', m.group(2))
+                    if re.match(r'^(.+)\^\{\}$', t): # detect tag "symlink"
+                        t = re.sub(r'\^\{\}$', '', t) # remove "symlink" chars, e.g. some-tag^{}
```

Comment:
try `git show-ref --tags --dereference`. A quote from stackoverflow:
_In git, a "normal" (annotated, not lightweight) tag is an object unto  itself, containing metadata and the SHA1 of the object it tags. There's  a pretty picture in the section of the git community book on the git object model (scroll to the bottom).  So, when you use show-ref on a normal tag, it will normally give you information about the tag object. With the -d/--dereference option, it will dereference the tag into the object the tag refers to, and provide information about it instead.  And a note on lightweight vs. annotated tags, in case you aren't aware of that: a lightweight tag is created by using git tag <tag name> (i.e. without any of the metadata-providing options like -a, -s, or -u).  It's not a tag object at all, just a ref *pointing straight to the  object you've tagged*. If you provide one of those options, you're  attaching metadata to the tag, so git creates a tag object to hold that._

Reply:
@screamerbg That tells me that some tags are references and some are tag objects. How do "symlinks" get involved here?